crates/matrix-sdk/tests/integration/room/spaces.rs

```diff
@@ -0,0 +1,315 @@
+                "example.org",
+                "other.example.org"
+            ]
+        },
+        "event_id": "$143273582443PhrSn:example.org",
+        "origin_server_ts": 1432735824653_u64,
+        "room_id": *DEFAULT_TEST_SPACE_ID,
+        "sender": "@example2:example.org", // Not equal to sender of m.room.space because
+                                           // equality is only required for power-levels
+        "state_key": *DEFAULT_TEST_ROOM_ID,
+        "type": "m.space.child",
+        "unsigned": {
+            "age": 1234
+        }
+    });
+
+    sync_space(&client, &server, sync_token, vec![child_event]).await;
+
+    let room = client.get_room(&DEFAULT_TEST_ROOM_ID).unwrap();
+
+    let spaces: Vec<ParentSpace> = room.parent_spaces().await.unwrap().collect().await;
+    assert_eq!(spaces.len(), 1);
+    assert_let!(Some(ParentSpace::Reciprocal(space)) = spaces.get(0));
+    assert_eq!(space.room_id(), *DEFAULT_TEST_SPACE_ID);
+}
+
+#[async_test]
+async fn parent_space_redacted_reciprocal() {
+    let (client, server) = logged_in_client().await;
+
+    let sync_token =
+        initial_sync_with_m_space_parent(&client, &server, &test_json::PARENT_SPACE_SYNC).await;
+
+    let child_event = json!({
+        "content": {},  // Redacted -> missing "via" key -> invalidates the relationship
+        "event_id": "$143273582443PhrSn:example.org",
+        "origin_server_ts": 1432735824653_u64,
+        "room_id": *DEFAULT_TEST_SPACE_ID,
+        "sender": "@example2:example.org",
+        "state_key": *DEFAULT_TEST_ROOM_ID,
+        "type": "m.space.child",
+        "unsigned": {
+            "age": 1234,
+            "redacted_by": "$15275047031IXQRi:localhost"
+        }
+    });
+
+    sync_space(&client, &server, sync_token, vec![child_event]).await;
+
+    let room = client.get_room(&DEFAULT_TEST_ROOM_ID).unwrap();
+
+    let spaces: Vec<ParentSpace> = room.parent_spaces().await.unwrap().collect().await;
+    assert_eq!(spaces.len(), 1);
+    assert_let!(Some(ParentSpace::Illegitimate(space)) = spaces.get(0));
+    assert_eq!(space.room_id(), *DEFAULT_TEST_SPACE_ID);
+}
+
+/// Adds @spaceadmin:localhost to the room with the given power level
+async fn setup_parent_member(
+    client: &Client,
+    server: &MockServer,
+    sync_token: String,
+    level: u64,
+) -> String {
+    let members = json!({
+        "chunk": [
+        {
+            "content": {
+                "avatar_url": null,
+                "displayname": "Space Administrator",
+                "membership": "join"
+            },
+            "event_id": "$151800140517rfvjc:localhost",
+            "membership": "join",
+            "origin_server_ts": 151800140,
+            "room_id": *DEFAULT_TEST_SPACE_ID,
+            "sender": "@spaceadmin:localhost",
+            "state_key": "@spaceadmin:localhost",
+            "type": "m.room.member",
+            "unsigned": {
+                "age": 2970366,
+            }
+        }
+        ]
```

Comment:
Indentation is weird here. Can you please either merge the outermost `{}` onto the same lines as the outermost `[]`, or indent the inner object once more?